blockmanager.go

```diff
@@ -347,21 +357,100 @@ func (b *blockManager) cfHandler() {
 					time.Sleep(QueryTimeout)
 				}
 			}
-			b.getAllCFHeaders(goodCheckpoints, store, fType)
-			mtxChkpt.Lock()
-			checkpoints[fType] = goodCheckpoints
-			mtxChkpt.Unlock()
-			wg.Done()
-		}(fType, store)
+
+			// Get all the headers up to the last known good
+			// checkpoint.
+			b.getCheckpointedCFHeaders(goodCheckpoints, store,
+				fType)
+
+			// Catch the filter headers up to the tip of the
+			// block header store and then stay caught up with
+			// each signal while watching for reorgs. We ignore
+			// checkpoints because at this point, we're past
+			// any valid checkpoints.
+			ticker := time.NewTicker(QueryTimeout)
+			defer ticker.Stop()
+			for {
```

Comment:
Seems like this won't actually exit? Or is the point that this runs in the background indefinitely after the initial set of checkpoints has been calculated?